File: 07_09_2024_Nth_Natural_Number.cpp
// User function Template for C++

class Solution {
  public:
    long long findNth(long long n) {
        // code here.
        if(n<9) return n;
        vector<int> temp;
        
        while(n>=9){
            int rm = n%9;
            temp.push_back(rm);
            n/=9;
            if(n<9){
                temp.push_back(n);
            }
        }
        long long ans =0;
        
        for(int i=temp.size()-1;i>=0;i--){
            ans=ans*10+temp[i];
        }
        
        return ans ;
    }
};

File: 07_09_2024_Nth_Natural_Number.py
#User function Template for python3
class Solution:
    def findNth(self, n):
        # code here
        if n < 9:
            return n
        
        temp = []
        
        while n >= 9:
            remainder = n % 9
            temp.append(remainder)
            n //= 9
            if n < 9:
                temp.append(n)
        
        ans = 0
        for i in range(len(temp) - 1, -1, -1):
            ans = ans * 10 + temp[i]
        
        return ans
